File: Algorithms/1275. Find Winner on a Tic Tac Toe Game/swift/solution.swift
class Solution {
    func tictactoe(_ moves: [[Int]]) -> String {
        var rows = [0, 0, 0]
        var cols = [0, 0, 0]
        var antidiag = 0
        var diag = 0

        for (index, move) in moves.enumerated() {
            let playerVal = index % 2 == 0 ? 1 : -1
            let row = move[0]
            let col = move[1]

            if row == col {
               diag += playerVal
            }

            if row + col == 2 {
               antidiag += playerVal
            }

            rows[row] += playerVal
            cols[col] += playerVal

            if diag == 3 || antidiag == 3 || rows[row] == 3 || cols[col] == 3 { return "A" }
            if diag == -3 || antidiag == -3 || rows[row] == -3 || cols[col] == -3 { return "B" }
        }

        if moves.count == 9 { return "Draw" }
        return "Pending"
    }
}

File: Algorithms/1275. Find Winner on a Tic Tac Toe Game/kotlin/solution.kt
class Solution {
    fun tictactoe(A: Array<IntArray>): String {
        val n = 3
        val rows = IntArray(n)
        val cols = IntArray(n)
        var diag = 0
        var secDiag = 0
        var isATurn = true
        var moves = 0
        for (row in A) {
            val (x, y) = row
            moves++
            if (isATurn) {
                rows[x]++
                cols[y]++
                if (x == y) diag++
                if (x + y == n - 1) secDiag++
            } else {
                rows[x]--
                cols[y]--
                if (x == y) diag--
                if (x + y == n - 1) secDiag--
            }
            if (rows[x] == n || cols[y] == n || diag == n || secDiag == n) return "A"
            else if (rows[x] == -n || cols[y] == -n || diag == -n || secDiag == -n) return "B"
            isATurn = isATurn.not()
        }
        if (moves == 9) return "Draw"
        return "Pending"
    }
}
